x-pack/plugins/code/public/sagas/breadcrumbs.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Action } from 'redux-actions';
+import { takeEvery } from 'redux-saga/effects';
+import chrome from 'ui/chrome';
+import { Match, routeChange } from '../actions';
+import { mainRoutePattern } from './patterns';
+
+function handleBreadCrumbs(action: Action<Match>) {
+  if (mainRoutePattern(action)) {
+    const { org, repo } = action.payload!.params;
+    chrome.breadcrumbs.set([
```

Comment:
I think this setup is good, but why do we do it in saga file, which is supposed to handle data fetch only.

Reply:
I think saga is designed for handling side effects.